.github/workflows/rspec_tests.yaml

```diff
@@ -0,0 +1,32 @@
+---
+name: RSpec test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  rspec_tests:
+    name: ${{ matrix.cfg.os }}(ruby ${{ matrix.cfg.ruby }})
+    strategy:
+      matrix:
+        cfg:
+          - {os: ubuntu-18.04, ruby: '2.5'}
+          - {os: ubuntu-18.04, ruby: '2.7'}
+          - {os: windows-2019, ruby: '2.5'}
+          - {os: windows-2019, ruby: '2.7'}
+
+    runs-on: ${{ matrix.cfg.os }}
+    steps:
+      - name: Checkout current PR
+        uses: actions/checkout@v2
```

Comment:
some of the puppet & facter versions of these use `fetch-depth: 0` like https://github.com/puppetlabs/puppet/blob/main/.github/workflows/checks.yaml#L25 I assume it limits the amount of git history we fetch, so maybe a good thing to add?